### PR TITLE
Create CosineWarmupScheduler class in benchmarks

### PR DESCRIPTION
**Pull request by Ra1Nik**
Add CosineWarmupScheduler class in benchmarks for imagenette

### CODE REVIEW

File: docs/source/getting_started/benchmarks/imagenette_benchmark.py [hunk 1]
```diff
@@ -237,6 +234,25 @@ def get_data_loaders(batch_size: int, model):
...
 
+class CosineWarmupScheduler(torch.optim.lr_scheduler.LambdaLR):
+    def __init__(self, optimizer, warmup_epochs, last_epoch=-1, verbose=False):
+        self.warmup_epochs = warmup_epochs
+        super().__init__(optimizer, self.scale_lr, last_epoch, verbose=False)
+
+    def scale_lr(self, epoch):
+        if epoch < self.warmup_epochs:
+            return epoch / self.warmup_epochs
+        else:
```
Comment (Ra1Nik) [inline docs/source/getting_started/benchmarks/imagenette_benchmark.py:245]: I could also do here:

`return 0.5*utils.cosine_schedule(epoch - self.warmup_epochs, max_epochs - self.warmup_epochs, 0, 1)`

But I think it hurts legibility

Comment (guarin) [reply docs/source/getting_started/benchmarks/imagenette_benchmark.py:245]: It is probably better to use the existing `cosine_schedule` method as it is well tested. Also I believe you don't need the `0.5` factor.

File: docs/source/getting_started/benchmarks/imagenette_benchmark.py [hunk 2]
```diff
@@ -237,6 +234,25 @@ def get_data_loaders(batch_size: int, model):
     return dataloader_train_ssl, dataloader_train_kNN, dataloader_test
 
 
+class CosineWarmupScheduler(torch.optim.lr_scheduler.LambdaLR):
```
Comment (guarin) [inline docs/source/getting_started/benchmarks/imagenette_benchmark.py:237]: As we use this class in many places it would maybe be better to move it into its own file. I would propose `lightly/utils/scheduler.py`. Could you also move the `cosine_schedule` method from `lightly/models/utils.py` there and update all the imports?

**Review by guarin — CHANGES_REQUESTED** [review]
Thanks a lot for the PR! This will make things a lot cleaner. Added some comments :)

File: tests/utils/test_scheduler.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import unittest
+import torch
+from torch import nn
+
+from lightly.utils import cosine_schedule
+from lightly.utils import CosineWarmupScheduler
+
+
+class TestModelUtils(unittest.TestCase):
```
Comment (guarin) [inline tests/utils/test_scheduler.py:9]:
```suggestion
class TestScheduler(unittest.TestCase):
```

File: tests/models/test_ModelUtils.py [hunk 4]
```diff
@@ -222,73 +230,56 @@ def _test_random_token_mask_parameters(self, device):
...
         assert candidate_maps_filtered.shape == (3, 1, 2)
-        assert candidate_maps_filtered.equal(torch.tensor([
-            [[1, 1]],
-            [[3, 3]],
-            [[1, 1]]
-        ]))
-
-
-    def test_cosine_schedule(self):
-        momentum_0 = cosine_schedule(1, 10, 0.99, 1)
```
Comment (guarin) [inline tests/models/test_ModelUtils.py:281]: Great that you moved the tests!

**Review by guarin — COMMENTED** [review]
Nice! Forgot something in the last review but we can merge this soon.

**Review by guarin — APPROVED** [review]
Awesome!
